src/nyx/output_2_csv.cpp

```diff
@@ -55,29 +55,39 @@ namespace Nyxus
 		return x;
 	}
 
+	std::string get_feature_output_fname (const std::string& intFpath, const std::string& segFpath)
+	{
+		std::string retval;
+		if (theEnvironment.separateCsv)
+		{
+			retval = theEnvironment.output_dir + "/_INT_" + getPureFname(intFpath) + "_SEG_" + getPureFname(segFpath) + ".csv";
+		}
+		else
+		{
+			retval = theEnvironment.output_dir + "/" + "NyxusFeatures.csv";
+		}
+		return retval;
+	}
+
+	const std::vector<std::string> mandatory_output_columns {"intensity_image", "segmentation_image", "ROI_label"};
+
 	// Saves the result of image scanning and feature calculation. Must be called after the reduction phase.
-	bool save_features_2_csv (std::string intFpath, std::string segFpath, std::string outputDir)
+	bool save_features_2_csv (const std::string & intFpath, const std::string & segFpath, const std::string & outputDir)
```

Comment:
Curious whether the arrow output code path needs anything similar to this. @JesseMckinzie